src/lib/netlink/vxlan.rs

```diff
@@ -40,111 +37,104 @@ const IFLA_VXLAN_GPE: u16 = 27;
+                }
+                IFLA_VXLAN_LIMIT => {
+                    info.max_address = parse_as_u32(nla.value());
+                }
+                IFLA_VXLAN_PORT_RANGE => {
+                    info.src_port_min = parse_as_be16(nla.value());
+                    info.src_port_max = parse_as_be16(&[nla.value()[2], nla.value()[3]]);
+                }
+                IFLA_VXLAN_PROXY => {
+                    info.proxy = parse_as_u8(nla.value()) > 0;
+                }
+                IFLA_VXLAN_RSC => {
+                    info.rsc = parse_as_u8(nla.value()) > 0;
+                }
+                IFLA_VXLAN_L2MISS => {
+                    info.l2miss = parse_as_u8(nla.value()) > 0;
+                }
+                IFLA_VXLAN_L3MISS => {
+                    info.l3miss = parse_as_u8(nla.value()) > 0;
+                }
+                IFLA_VXLAN_PORT => {
+                    info.dst_port_min = parse_as_be16(nla.value());
+                    info.dst_port_max = parse_as_be16(&[nla.value()[2], nla.value()[3]]);
+                }
+                IFLA_VXLAN_GROUP6 => {
+                    info.remote = parse_as_ipv6(nla.value());
+                }
+                IFLA_VXLAN_LOCAL6 => {
+                    info.local = parse_as_ipv6(nla.value());
+                }
+                IFLA_VXLAN_UDP_CSUM => {
+                    info.udp_check_sum = parse_as_u8(nla.value()) > 0;
+                }
+                IFLA_VXLAN_UDP_ZERO_CSUM6_TX => {
+                    info.udp6_zero_check_sum_tx = parse_as_u8(nla.value()) > 0;
+                }
+                IFLA_VXLAN_UDP_ZERO_CSUM6_RX => {
+                    info.udp6_zero_check_sum_rx = parse_as_u8(nla.value()) > 0;
+                }
+                IFLA_VXLAN_REMCSUM_TX => {
+                    info.remote_check_sum_tx = parse_as_u8(nla.value()) > 0;
+                }
+                IFLA_VXLAN_REMCSUM_RX => {
+                    info.remote_check_sum_rx = parse_as_u8(nla.value()) > 0;
+                }
+                IFLA_VXLAN_GBP => {
+                    info.gbp = true;
+                }
+                IFLA_VXLAN_REMCSUM_NOPARTIAL => {
+                    info.remote_check_sum_no_partial = true;
+                }
+                IFLA_VXLAN_COLLECT_METADATA => {
+                    info.collect_metadata = parse_as_u8(nla.value()) > 0;
+                }
+                IFLA_VXLAN_LABEL => {
+                    info.label = parse_as_be32(nla.value());
+                }
+                IFLA_VXLAN_GPE => {
+                    info.gpe = true;
+                }
+                IFLA_VXLAN_TTL_INHERIT => {
+                    info.ttl_inherit = parse_as_u8(nla.value()) > 0;
+                }
+                IFLA_VXLAN_DF => {
+                    info.df = parse_as_u8(nla.value());
+                }
+                _ => eprintln!("Unknwon nla_type {}, {:?}", nla.kind(), nla.value()),
```

Comment:
`eprintln!("Unhandled VxLAN IFLA_INFO_DATA: {}, {:?}", nla.kind(), nla.value())`

Reply:
We need more context when the error happens as many InfoData are sharing the same value of `nla.kind()` in kernel.